iroh-resolver/src/resolver.rs

```diff
@@ -693,13 +681,15 @@ impl LoaderContext {
 
 impl Drop for LoaderContext {
     fn drop(&mut self) {
-        if Arc::strong_count(&self.inner) == 1 {
+        let count = Arc::strong_count(&self.inner);
+        debug!("session {} dropping loader context {}", self.id, count);
+        if count == 1 {
             if let Err(err) = self
                 .inner
                 .try_lock()
                 .expect("last reference, no lock")
                 .closer
-                .try_send(self.id)
+                .send_blocking(self.id)
```

Comment:
I assume `drop` is called in a separate task? Blocking is scary with tokio, on connectivity issues this might block other tasks.

Reply:
nope it actually blocks 😓  because we can't do any a sync work in drop, or tokio panics